src/widget/wcuemenupopup.cpp

```diff
@@ -0,0 +1,129 @@
+#include <QHBoxLayout>
+#include <QInputDialog>
+#include <QVBoxLayout>
+
+#include "engine/engine.h"
+#include "util/color/color.h"
+
+WCueMenuPopup::WCueMenuPopup(QWidget* parent)
+        : QWidget(parent) {
+    QWidget::hide();
+    setWindowFlags(Qt::Popup);
+    setObjectName("WCueMenuPopup");
+
+    m_pCueNumber = new QLabel(this);
+    m_pCueNumber->setToolTip(tr("Cue number"));
+    m_pCueNumber->setObjectName("CueNumberLabel");
+    m_pCueNumber->setAlignment(Qt::AlignLeft);
+
+    m_pCuePosition = new QLabel(this);
+    m_pCuePosition->setToolTip(tr("Cue position"));
+    m_pCuePosition->setObjectName("CuePositionLabel");
+    m_pCuePosition->setAlignment(Qt::AlignRight);
+
+    m_pEditLabel = new QLineEdit(this);
+    m_pEditLabel->setToolTip(tr("Edit cue label"));
+    m_pEditLabel->setObjectName("CueLabelEdit");
+    connect(m_pEditLabel, &QLineEdit::textEdited, this, &WCueMenuPopup::slotEditLabel);
+    connect(m_pEditLabel, &QLineEdit::returnPressed, this, &WCueMenuPopup::hide);
+
+    m_pColorPicker = new WColorPicker(this);
+    m_pColorPicker->setObjectName("CueColorPicker");
+    connect(m_pColorPicker, &WColorPicker::colorPicked, this, &WCueMenuPopup::slotChangeCueColor);
+
+    m_pDeleteCue = new QPushButton("", this);
+    m_pDeleteCue->setToolTip(tr("Delete this cue"));
+    m_pDeleteCue->setObjectName("CueDeleteButton");
+    connect(m_pDeleteCue, &QPushButton::clicked, this, &WCueMenuPopup::slotDeleteCue);
+
+    QHBoxLayout* pLabelLayout = new QHBoxLayout();
+    pLabelLayout->addWidget(m_pCueNumber);
+    pLabelLayout->addStretch(1);
+    pLabelLayout->addWidget(m_pCuePosition);
+
+    QVBoxLayout* pLeftLayout = new QVBoxLayout();
+    pLeftLayout->addLayout(pLabelLayout);
+    pLeftLayout->addWidget(m_pEditLabel);
+    pLeftLayout->addWidget(m_pColorPicker);
+
+    QVBoxLayout* pRightLayout = new QVBoxLayout();
+    pRightLayout->addWidget(m_pDeleteCue);
+    pRightLayout->addStretch(1);
+
+    QHBoxLayout* pMainLayout = new QHBoxLayout();
+    pMainLayout->addLayout(pLeftLayout);
+    pMainLayout->addSpacing(5);
+    pMainLayout->addLayout(pRightLayout);
+    setLayout(pMainLayout);
+}
+
+void WCueMenuPopup::setTrackAndCue(TrackPointer pTrack, CuePointer pCue) {
+    if (pTrack && pCue) {
+        m_pTrack = pTrack;
+        m_pCue = pCue;
+
+        int hotcueNumber = m_pCue->getHotCue();
+        QString hotcueNumberText = "";
+        if (hotcueNumber != Cue::kNoHotCue) {
+            // Programmers count from 0, but DJs count from 1
+            hotcueNumberText = QString(tr("Hotcue #%1")).arg(QString::number(hotcueNumber + 1));
+        }
+        m_pCueNumber->setText(hotcueNumberText);
+
+        QString positionText = "";
+        double startPosition = m_pCue->getPosition();
+        double endPosition = m_pCue->getEndPosition();
+        if (startPosition != Cue::kNoPosition) {
+            positionText = mixxx::Duration::formatTime(startPosition / m_pTrack->getSampleRate() / mixxx::kEngineChannelCount);
```

Comment:
I think we can effort here the same accuracy as in the deck's display.

12:05.78

Reply:
OK.